### PR TITLE
Refactor/module name

### PR DESCRIPTION
**Pull request by doronz88**

### CODE REVIEW

File: xontrib/branch_expansion.py [hunk 1]
```diff
@@ -32,7 +33,11 @@ def expand_braces(cmd, **_):
                 prefix + option + postfix for option in options.split(",")
             ]
             expanded_cmds.extend(expanded_parts)
+            requires_expansion = True
```
Comment (rpdelaney) [inline xontrib/branch_expansion.py]: Why not just return here? No need to wait

**Review by rpdelaney — CHANGES_REQUESTED** [review]
I don't see how this restructures the module name. Do you have unpushed changes?

**Comment by doronz88** [comment]
> I don't see how this restructures the module name. Do you have unpushed changes?

Renamed the python module to be imported via `xontrib load` (look at the first commit)

File: xontrib/branch_expansion.py (hunk 1 above)
Comment (doronz88) [reply xontrib/branch_expansion.py]: the `{}` can exist in more parts

**Comment by rpdelaney** [comment]
Ah, looks like you renamed it to `branch_expansion` but it should be `brace_expansion`

File: xontrib/branch_expansion.py (hunk 1 above)
Comment (rpdelaney) [inline xontrib/branch_expansion.py]:
```suggestion
            return " ".join(expanded_cmds)
```

File: xontrib/branch_expansion.py [hunk 2]
```diff
@@ -32,7 +33,11 @@ def expand_braces(cmd, **_):
...
         else:
             expanded_cmds.append(part)
 
-    return " ".join(expanded_cmds)
+    if requires_expansion:
+        return " ".join(expanded_cmds)
+
+    return cmd
```
Comment (rpdelaney) [inline xontrib/branch_expansion.py:43]:
```suggestion
```

Comment (rpdelaney) [inline xontrib/branch_expansion.py]:
```suggestion
```

File: xontrib/branch_expansion.py [hunk 3]
```diff
@@ -23,6 +23,7 @@
 def expand_braces(cmd, **_):
     parts = cmd.split()
     expanded_cmds = []
+    requires_expansion = False
```
Comment (rpdelaney) [inline xontrib/branch_expansion.py]:
```suggestion
```

File: xontrib/branch_expansion.py (hunk 1 above)
Comment (doronz88) [reply xontrib/branch_expansion.py]: but that way i don't know if there are parts with both `{` and `}` so i can stop the expansion

File: xontrib/branch_expansion.py (hunk 2 above)
Comment (doronz88) [reply xontrib/branch_expansion.py]: as i said previosly, this is needed

File: xontrib/brace_expansion.py [hunk 4]
```diff
@@ -21,8 +21,12 @@
 
 @events.on_transform_command
 def expand_braces(cmd, **_):
+    if '[' in cmd or '$(' in cmd or '@(' in cmd:
+        return cmd
+
```
Comment (rpdelaney) [inline xontrib/brace_expansion.py:26]: Please don't sprawl the scope of the PR. I don't like this and won't accept it; whatever this is doing should be done with an ast, but more importantly, it should be another PR so we can discuss that separately.

File: xontrib/brace_expansion.py [hunk 5]
```diff
@@ -21,8 +21,12 @@
...
     parts = cmd.split()
     expanded_cmds = []
+    requires_expansion = False
 
     for part in parts:
         if "{" in part and "}" in part:
```
Comment (rpdelaney) [inline xontrib/brace_expansion.py:32]: A regular expression might be more reliable and could exclude nested braces. Could be like this:

```python
import re

pattern = r"\{([a-zA-Z],?)+\}"
test_strings = ["{a,o,e,u}", "{x,y,z}", "{aoeu}", "{a,o,e,u,{x},y}"]

for test in test_strings:
    if re.match(pattern, test):
        print(f"Match: {test}")
    else:
        print(f"No match: {test}")
```

File: xontrib/brace_expansion.py [hunk 6]
```diff
@@ -32,7 +36,11 @@ def expand_braces(cmd, **_):
...
             expanded_cmds.extend(expanded_parts)
+            requires_expansion = True
         else:
             expanded_cmds.append(part)
 
-    return " ".join(expanded_cmds)
+    if requires_expansion:
+        return " ".join(expanded_cmds)
+
+    return cmd
```
Comment (rpdelaney) [inline xontrib/brace_expansion.py:46]: If we reach line 39, we always end up returning from line 44. There is no reachable code between these two points. Please remove `requires_expansion`.

Comment (rpdelaney) [reply xontrib/brace_expansion.py:46]: It would really be better to change one thing at a time to avoid having the core issue bogged down.

**Review by rpdelaney — CHANGES_REQUESTED** [review]
